analyzers/src/SonarAnalyzer.Utilities/RuleDetailBuilder.cs

```diff
@@ -33,177 +33,63 @@ namespace SonarAnalyzer.Utilities
-        };
-
-        private static readonly Assembly SonarAnalyzerUtilitiesAssembly = typeof(RuleDetailBuilder).Assembly;
-
         public static IEnumerable<RuleDetail> GetAllRuleDetails(AnalyzerLanguage language)
         {
-            return new RuleFinder()
-                .GetAnalyzerTypes(language)
-                .SelectMany(t => GetRuleDetailFromRuleAttributes(t, language));
-        }
-
-        private static IEnumerable<RuleDetail> GetRuleDetailFromRuleAttributes(Type analyzerType,
-            AnalyzerLanguage language)
-        {
-            return analyzerType.GetCustomAttributes<RuleAttribute>()
-                .Select(ruleAttribute => GetRuleDetail(ruleAttribute, analyzerType, language));
-        }
-
-        private static RuleDetail GetRuleDetail(RuleAttribute rule, Type analyzerType, AnalyzerLanguage language)
-        {
-            var resources = new ResourceManager("SonarAnalyzer.RspecStrings", analyzerType.Assembly);
-
-            var ruleDetail = new RuleDetail
+            var resources = language.LanguageName switch
             {
-                Key = rule.Key,
-                Type = GetBackwardsCompatibleType(resources.GetString($"{rule.Key}_Type")),
-                Title = resources.GetString($"{rule.Key}_Title"),
-                Severity = resources.GetString($"{rule.Key}_Severity"),
-                Status = resources.GetString($"{rule.Key}_Status"),
-                IsActivatedByDefault = bool.Parse(resources.GetString($"{rule.Key}_IsActivatedByDefault")),
-                Description = GetResourceHtml(rule, language),
-                Remediation = ToSonarQubeRemediationFunction(resources.GetString($"{rule.Key}_Remediation")),
-                RemediationCost = resources.GetString($"{rule.Key}_RemediationCost")
+                LanguageNames.CSharp => new ResourceManager("SonarAnalyzer.RspecStrings", typeof(Rules.CSharp.FlagsEnumZeroMember).Assembly),
+                LanguageNames.VisualBasic => new ResourceManager("SonarAnalyzer.RspecStrings", typeof(Rules.VisualBasic.FlagsEnumZeroMember).Assembly),
+                _ => throw new InvalidOperationException("Unexpected language: " + language)
             };
-
-            ruleDetail.Tags.AddRange(resources.GetString($"{rule.Key}_Tags").Split(new[] { ',' },
-                StringSplitOptions.RemoveEmptyEntries));
-
-            GetParameters(analyzerType, ruleDetail);
-            GetCodeFixNames(analyzerType, ruleDetail);
-
-            return ruleDetail;
+            return new RuleFinder()
+                .GetAnalyzerTypes(language)
+                .Select(x => (DiagnosticAnalyzer)Activator.CreateInstance(x))
+                .SelectMany(x => UniqueRuleIds(x).Select(id => new { Id = id, Type = x.GetType() }))
+                .GroupBy(x => x.Id)    // Same ruleId can be in multiple classes (see InvalidCastToInterface)
+                .Select(x => CreateRuleDetail(language, resources, x.Key, x.Select(item => item.Type)));
```

Comment:
This file redesign is tricky. Intermediate changes failed Java UTs and ITs.
**Before:**
When a rule is in multiple classes (like `InvalidCastToInterface`), `RuleAttribute` is only on one of them. Rule parameters and rule code fixes are registered only from this class. The other is ignored.

**After:**
Group `<id, class>` pair by `<id>`. Create single `RuleDetail` from it.
Iterate all `<class>` types. Fill parameters and code fixes.